src/installer/tests/HostActivation.Tests/NativeHosting/ComponentActivation.cs

```diff
@@ -211,6 +233,7 @@ public class SharedTestState : SharedTestStateBase
             public string DotNetRoot { get; }
 
             public TestProjectFixture ComponentWithNoDependenciesFixture { get; }
+            public TestProjectFixture SelfContainedComponentWithNoDependenciesFixture { get; }
```

Comment:
Let's not use "self-contained component" since we don't really support that.
Also we're using SDK behavior which is "unsupported" and probably relatively flaky - building classlib with self-contained true.

I would just create a new test application (if we can't use the existing ones), build it as self-contained and use that - the test validates we can load the app as an app and then get a function pointer from it. 

I want to avoid confusing in the future when somebody reads the tests and they look like we support self-contained components, when we really don't.

Reply:
I'm sorry, but I'm going to have to push back on this. There's nothing flaky about using `--self-contained` with `EnableDynamicLoading=True`. It works.

Reply:
"Works on a sample project" != "works always". And even if it did, it's a scenario which is missing several important things:
* Nobody actually went through the SDK and made sure that it "should" work
* There's no tests for it in the SDK or the runtime
* There are cases where it doesn't work outside of SDK (loading such component via `AssemblyDependencyResolver` will do really weird and "bad" things)
* There's no documentation that it should work and if so how
* We don't support it (which is not just about it working, but about us signing up to make sure that it's going to continue to work in future releases, fixing bugs in the existing ones and so on)

The reason I don't want this in the tests is that by having a test for it, it's basically saying "we want to support this". But doing that would require lot more work, which we're not planning to do now.

Reply:
I will change the project that this test uses. I will also create an issue in dotnet/sdk for official support on `--self-contained` combined with `EnableDynamicLoading=True`. I would like to point out that this is **not** necessarily a component. This would be the supported way to build this hybrid app/component.

I don't understand why the runtime needs to add support for this combination. You need to write your own native host to start the runtime for this, but that's the whole point. The only difference between publishing with the currently supported way of publishing self-contained with an executable and this "new" way of `EnableDynamicLoading=True` will be the presence of the .exe file. The hybrid's dll still needs to be there just like the .exe way, and the  `.runtimeconfig.json` and the `.deps.json` will (need to) match the .exe way. `AssemblyDependencyResolver` will have just as much problems with a project that was published the .exe way.

Reply:
The main reason I'm trying to make a clear distinction between app and component is that that's how SDK and docs and most users see it: app versus classlib. I do agree with all your points, I don't think the problem here is that things won't work in the hybrid scenario (since as you pointed out, it sort of doesn't matter if it's app or classlib). I just think that we need to get the terminology right - so that it leads people to solutions which are likely to work even in other scenarios. Currently if I build something as classlib all our docs and terminology lead me to believe that I can use it as both native (native hosting APIs) and managed (`AssemblyDependencyResolver`) plugin - if we start officially talking about self-contained components, then I would naturally think that it also works in all these cases. So it's mostly about the right perception. I've seen it too many times that sloppy naming or too vague guidance leads people to solutions which work as a demo, but then break for real apps on some more complex cases. I think it's better if users find the proven mainstream solutions first and only venture into the more complicated ones when absolutely necessary.

So from that point of view calling `AsssemblyDependencyResolver` on an app is currently simply not supported - we never mention it anywhere. And while it might work in some cases,  we don't want people to do it, since they may get into problems.

>  I would like to point out that this is not necessarily a component. This would be the supported way to build this hybrid app/component.

The problem is that SDK currently only knows about apps or "components" - nothing in between. So the support for --self-contained on classlib projects is essentially support for self-contained components. You may not end up using it as a pure component, but from SDKs point of view it has to look like one. Alternatively we would have to introduce a new concept - some "hybrid" thing.

I'm not trying to shut off special cases - I'm happy if we can make the product work in new interesting ways, but there has to be a distinction between "The recommended proven way of doing things" and "Going off the rails a little bit into this expert level area". I know it sounds like playing with words, but experience tells me it's important.

Reply:
Yes, sorry, I was lazy with my naming in the tests and it needed to change.

I'm not sure why we're continuing the SDK discussion here instead of in the [issue](https://github.com/dotnet/sdk/issues/12043)? Like I've said before this is not a component and not an app. We need to stop trying to pretending that it's one or other, it's a new concept.